### ISSUE TITLE
Cannot distinguish types by literal field if one of them is None

### ISSUE THREAD

**Issue by Nhfiam** [microsoft/pyright#7590]
Expected behavior: no pyright errors
Actual behavior: `error: Cannot access member "x" for type "T2"`

**Code**
```python
import typing


class T1:
    status_code: typing.Literal[1] = 1
    x: int = 0


class T2:
    status_code: None = None


def f(t: T1 | T2) -> int:
    if t.status_code == 1:
        return t.x
    return 0
```

**Version**
`pyright 1.1.356`, command-line tool

**Comment by Nhfiam** [comment]
Fun fact: if you replace `status_code: None` with `status_code: typing.Literal[2]` then everything works as expected

**Comment by erictraut** [comment]
Pyright is working as designed here, so I don't consider this a bug. 

For a full list of supported type guard patterns, see [this documentation](https://microsoft.github.io/pyright/#/type-concepts-advanced?id=type-guards). It appears that you're attempting to use the pattern "`x.y == L` and `x.y != L` (where L is a literal expression and x is a type that is distinguished by a field or property with a literal type)", but in this case the two types `T1` and `T2` are not distinguished by a literal value. One of them is using `None` instead.

I've confirmed that mypy's handling of this type guard pattern does support `None`. That's evidence that this pattern is common, so I'm willing to extend the support here to include `None`.

I'm changing this from a bug to an enhancement request.

**Comment by erictraut** [comment]
This will be included in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.357, which I just published.
